spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/MappingRelationalConverter.java

```diff
@@ -606,34 +608,65 @@ private boolean shouldReadEmbeddable(ConversionContext context, RelationalPersis
 		return false;
 	}
 
+	/**
+	 * Read and convert a single value that is comming from a database to the {@literal targetType} expected by the domain
+	 * model.
+	 * 
+	 * @param value a value as it is returned by the driver accessing the persistence store. May be {@code null}.
+	 * @param targetType {@link TypeInformation} into which the value is to be converted. Must not be {@code null}.
+	 * @return
+	 */
 	@Override
 	@Nullable
-	public Object readValue(@Nullable Object value, TypeInformation<?> type) {
+	public Object readValue(@Nullable Object value, TypeInformation<?> targetType) {
 
 		if (null == value) {
 			return null;
 		}
 
-		return getPotentiallyConvertedSimpleRead(value, type);
+		TypeInformation<?> originalTargetType = targetType;
+		value = readTechnologyType(value);
```

Comment:
These concerns can easily be handled in `JdbcConverter` and `R2dbcConverter`. Risking code duplications is less of an concern than introducing new conceptual methods. Technology-type and module-type aren't common across other data modules.

Reply:
Done. At least I hope I understood your intention correctly.